components/login/LoginForm.js

```diff
@@ -1,13 +1,59 @@
+import { useState } from 'react';
+import {
+    getAuth,
+    signInWithEmailAndPassword
+} from 'firebase/auth';
+import { firebaseApp } from '../../config/firebaseConfig';
 import EmailInputWithLabel from './EmailInputWithLabel';
 import PasswordInputWithLabel from './PasswordInputWithLabel';
 import SubmitButton from './SubmitButton';
 
-const LoginForm = () => (
-    <>
-        <EmailInputWithLabel />
-        <PasswordInputWithLabel />
-        <SubmitButton />
-    </>
-);
+const LoginForm = () => {
+
+    const auth = getAuth(firebaseApp)
+
+    const [loginEmail, setLoginEmail] = useState('');
+    const [loginPassword, setLoginPassword] = useState('');
+    const [loginNotification, setLoginNotification] = useState('');
+
+    const handleEmail = (newValue) => {
+        setLoginEmail(newValue);
+    };
+
+    const handlePassword = (newValue) => {
+        setLoginPassword(newValue);
+    };
+
+    const handleLogin = (e) => {
+        e.preventDefault();
+        signInWithEmailAndPassword(auth, loginEmail, loginPassword)
+            .catch((err) => {
+                console.log(err.code, err.message)
+                setLoginNotification(err.message)
+                // setTimeout(() => {
+                //     setLoginNotification('')
+                // }, 2000)
+            })
+        setLoginEmail('')
+        setLoginPassword('')
+    }
+
+    return (
+        <>
+            <div>
+                <h1>Log in</h1>
+                <form onSubmit={handleLogin}>
+                    <EmailInputWithLabel value={loginEmail} onChange={handleEmail} />
+                    <br />
+                    <PasswordInputWithLabel value={loginPassword} onChange={handlePassword} labelTextContent='Password' />
```

Comment:
You don't have to pass a prop to a component to give it a specific text content. That's what React children props are for. Just make the component take the text before a closing tag, i.e., <PassWordInputWithLabel>Password</PasswordInputWithLabel> and it's done and over without having to bother with passing props. Same can be done with all other reusables, e.g., button, email input.

Reply:
Tried it with the following snippets:
```
                <PasswordInputWithLabel value={signupPassword} onChange={handlePassword}>Password</PasswordInputWithLabel>
                
                <SubmitButton>Sign up</SubmitButton>
```
No text was rendered. Also, I was going to offer to have the input labels styled as hidden having them only for accessibility machine reading purposes and making visible inputs placeholders instead.